File: android/GoCoRo/app/src/main/java/com/wcare/android/gocoro/http/KnowledgeMessage.java
package com.wcare.android.gocoro.http;

/**
 * Created by ttonway on 2017/2/23.
 */
public class KnowledgeMessage {

    public String title;
    public String posterUrl;
    public String description;
    public String url;
    public long createdAt;

    @Override
    public String toString() {
        return "KnowledgeMessage{" +
                "title='" + title + '\'' +
                ", posterUrl='" + posterUrl + '\'' +
                ", description='" + description + '\'' +
                ", url='" + url + '\'' +
                ", createdAt=" + createdAt +
                '}';
    }
}

File: web/gocoro-web/models/knowledge_message.js
module.exports = function (orm, db) {
    var Message = db.define('knowledge_message', {
            title: {type: 'text', required: true},
            posterUrl: {type: 'text'},
            description: {type: 'text'},
            url: {type: 'text', required: true},
            createdAt: {type: 'date', required: true, time: true}
        },
        {
            methods: {
                serialize: function () {
                    return {
                        title: this.title,
                        posterUrl: this.posterUrl,
                        description: this.description,
                        url: this.url,
                        createdAt: this.createdAt.valueOf()
                    }
                }
            }
        });
    return Message;
};
